### PR TITLE
MABR and DMBR ZO

### PR DESCRIPTION
**Pull request by luohezhiming**
## Summary/Motivation:

Add mabr and dmbr models for 1598_Skerlos case study

## Changes proposed in this PR:
- add mabr and dmbr model
- add corresponding tests

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/core/zero_order_costing.py [hunk 1]
```diff
@@ -661,6 +663,58 @@ def cost_deep_well_injection(blk):
...
+        blk.capital_cost = pyo.Var(
+            initialize=1,
+            units=blk.config.flowsheet_costing_block.base_currency,
+            bounds=(0, None),
+            doc="Capital cost of unit operation")
+
+        Q_treated = pyo.units.convert(blk.unit_model.properties_treated[t0].flow_vol,
+                              to_units=pyo.units.L / pyo.units.hour)
+
+        ES_reactor = (1/pyo.units.convert(A, to_units=pyo.units.L / pyo.units.m**2 / pyo.units.hour))*Q_treated
```
Comment (andrewlee94) [inline watertap/core/zero_order_costing.py]: A minor comment, but I think you could avoid a unit conversion by replacing lines 696 to 699:

```
ES_reactor = pyo.units.convert(blk.unit_model.properties_treated[t0].flow_vol/A, to_units=pyo.units.m**2)
```

File: watertap/core/zero_order_costing.py [hunk 2]
```diff
@@ -661,6 +663,58 @@ def cost_deep_well_injection(blk):
...
+
+        expr = pyo.units.convert(ES_reactor*B, to_units=blk.config.flowsheet_costing_block.base_currency)
```
Comment (andrewlee94) [inline watertap/core/zero_order_costing.py]: You could save another unit conversion here by combining this with the one above.

```
expr = pyo.units.convert(
    blk.unit_model.properties_treated[t0].flow_vol/A*B,
    to_units=blk.config.flowsheet_costing_block.base_currency)
```

File: watertap/core/zero_order_costing.py [hunk 3]
```diff
@@ -752,6 +806,72 @@ def cost_gac(blk):
...
+
+        SV_reactor = Q_treated*ammonium_reacted
+
+        ES_reactor = (1/pyo.units.convert(A, to_units=pyo.units.kg / pyo.units.m**2 / pyo.units.hour))*SV_reactor
+
+        DCC_reactor = pyo.units.convert(ES_reactor*B, to_units=blk.config.flowsheet_costing_block.base_currency)
+
+        ES_blower = C*pyo.units.convert(blk.unit_model.blower_size, to_units=pyo.units.m ** 2)
+
+        DCC_blower = pyo.units.convert(ES_blower*D, to_units=blk.config.flowsheet_costing_block.base_currency)
```
Comment (andrewlee94) [inline watertap/core/zero_order_costing.py]: As above, you can save some unit conversion operations by combining many of these. In the end, I think you should only need two unit conversions.

File: watertap/examples/case_studies/wastewater_resource_recovery/case_1598.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    units: dimensionless
+  TPEC:
+    # Total Purchased Equipment Cost
+    value: 3.4
+    units: dimensionless
+  TIC:
+    # Total Installed Cost
+    value: 1.65
+    units: dimensionless
+# currency_definitions:
```
Comment (andrewlee94) [inline watertap/examples/case_studies/wastewater_resource_recovery/case_1598.yaml]: You should remove this comment as it is no longer needed. I am guessing this is also in the default file as well, so it should be removed there too.

**Review by andrewlee94 — CHANGES_REQUESTED** [review]
A couple of minor comments, most about reducing overhead by combining expressions to reduce the number of unit conversion calls.

**Comment by luohezhiming** [comment]
> A couple of minor comments, most about reducing overhead by combining expressions to reduce the number of unit conversion calls.

Thank you for all your suggestions, I've refined the code accordingly to avoid multiple unit conversions.

**Review by andrewlee94 — DISMISSED** [review]

**Review by adam-a-a — DISMISSED** [review]
LGTM. Just a quick note: there are a few comments that I originally entered in the yaml files that were kept in here (some of them still apply, while others may not). We can revisit/cleanup at a later point though.

**Comment by luohezhiming** [comment]
> LGTM. Just a quick note: there are a few comments that I originally entered in the yaml files that were kept in here (some of them still apply, while others may not). We can revisit/cleanup at a later point though.

@andrewlee94 @adam-a-a I just cleaned up those deprecated comments in the yaml file.

**Review by andrewlee94 — APPROVED** [review]

**Review by adam-a-a — APPROVED** [review]
LGTM

**Pull request merged**